### PR TITLE
PR

### LINKED ISSUE #1: Bot for filling forms from messenger
We are the team of corporate messenger [dialog](http://www.dlg.im). We have Java Bot SDK to create bots in our messenger. Since it is new, we want to order a few bots from the community for demonstration purposes.       
  
The documentation of our Java Bot SDK is here: https://docs.dlg.im/bots/java-bot.html    
  
Our messenger has a web client, you should use it for the development:     https://grpc-test.transmit.im/    
For web use endpoint: https://grpc-test.transmit.im:8443    
For grpc use endpoint: https://grpc-test.transmit.im:9443
Test accounts:    
alice / password    

After you've logged in you should write to  `Security Bot` user  and write command `/bot new bot_username bot_name` where `bot_username` is the username you can choose for your new bot and `bot_name` is his display name
  
We've open-sourced the bot for stress tests. This bot is able to register on the server, synchronizing with the phonebook and sending messages.    
The bot is available at https://github.com/dialogs/stressbot    
  
Please feel free to ask questions in our community https://dlg.im/@botsfordialog     
Our developers are always happy to help and answer your questions.     
   
Scenario for the bot:     
This bot's purpose is to fill in a table in Google Sheets the same way it is filled in with a Google Form. In this example, we will set up bot to maintain a list of purchases in our office. To submit a bounty, bot must be connected to any public Google Sheet.  
  
` The very first message from the bot to the user `  
Bot: Hi! Send me what you need in the office and I will create the purchase request.  
` Each user request will be a table's row, answers to the bot questions will be columns of the table. `  
User: Coca-Cola  
` Bot creates a new row in the table. The user's request is in column A (column title - "Order"). User login is in column B (column title - "Customer"). The time of the order is kept in column C (column title - "Order Time"). `  
Bot: Is it food or not food?  
Bot: [dropdown menu with a choice]: -Food; -Not food  
` User response is kept in column D (column title - "Order type") `  
Bot: In which office is this order?  
Bot: [dropdown menu with a choice]: -DF; -Pok; -Nov; -Znam.  
` User’s answer is kept in column E (column title - "Order office") `  
Bot: Thank you, your order is placed and will be purchased soon.  
The office administrator will contact you as soon as the order arrives at the office.  
  
For bounty to be verified, bot must be started and work on https://grpc-test.transmit.im/ and should fill in any table in Google Sheets

### PR DESCRIPTION
**Pull request by m8928**
Fixes #1

### CODE REVIEW

**Comment by Maay** [comment]
@m8928 please push us only the bot code, not the whole sdk.

**Comment by m8928** [comment]
Code Cleanning.

File: src/main/java/im/dlg/botsdk/formfillingbot/FormfillingBot.java [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+                    }
+
+                    if (orderList.isComplete()) {
+                        bot.messaging().send(event.getPeer(), orderList.getCompleteMessage());
+
+                        try {
+                            GoogleSheet.updateSpreadSheet(orderListMap.get(senderId).getSheetData());
+                        }
+                        catch (Exception e) {
+                            e.printStackTrace();
```
Comment (nkt) [inline src/main/java/im/dlg/botsdk/formfillingbot/FormfillingBot.java]: You should tell user that action was failed.

File: src/main/java/im/dlg/botsdk/formfillingbot/FormfillingBot.java [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+        );
+
+        bot.await();
+    }
+
+    public static OrderList createOrderForm(String user) {
+        Order first = new Order("Hi! Send me what you need in the office and I will create the purchase request.");
+
+        List<InteractiveSelectOption> secondSelectOptions = new ArrayList<>();
+        secondSelectOptions.add(new InteractiveSelectOption("Food", "Food"));
```
Comment (nkt) [inline src/main/java/im/dlg/botsdk/formfillingbot/FormfillingBot.java]: Move form description into config file.
Something like this:
```json
{
	"welcome": "Hi! Send me what you need in the office and I will create the purchase request.",
	"survey": [
		{
			"type": "select",
			"title": "Is it food?",
			"options": ["Food", "Not food"]
		},
		{
			"type": "text",
			"title": "Product link"
		},
		....
	]
}
```

File: src/main/java/im/dlg/botsdk/formfillingbot/FormfillingBot.java [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ExecutionException;
+
+public class FormfillingBot {
+    public static void main(String[] args) throws InterruptedException, ExecutionException {
+
+        final String OrderStartMessage = "Hey";
+        ConcurrentHashMap<Integer, OrderList> orderListMap = new ConcurrentHashMap<>();
+
+        Bot bot = Bot.start("df8c6600c2a45b711e097fd9e5e7f3cfee025e41").get();
```
Comment (nkt) [inline src/main/java/im/dlg/botsdk/formfillingbot/FormfillingBot.java]: This should be configured

File: src/main/java/im/dlg/botsdk/formfillingbot/GoogleSheet.java [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+                .setDataStoreFactory(new FileDataStoreFactory(new java.io.File(TOKENS_DIRECTORY_PATH)))
+                .setAccessType("offline")
+                .build();
+        return new AuthorizationCodeInstalledApp(flow, new LocalServerReceiver()).authorize("user");
+    }
+
+    public static void updateSpreadSheet(List<Object> array) throws Exception {
+        final NetHttpTransport HTTP_TRANSPORT = GoogleNetHttpTransport.newTrustedTransport();
+        final String spreadsheetId = "1fLBMr6hGuCQAGQcljKH_UKaVJBUWm7PH6nEzIjNVla4";
+        final String range = "Sheet1!A:E";
```
Comment (nkt) [inline src/main/java/im/dlg/botsdk/formfillingbot/GoogleSheet.java]: This should be configured

File: src/main/java/im/dlg/botsdk/formfillingbot/GoogleSheet.java [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+                HTTP_TRANSPORT, JSON_FACTORY, clientSecrets, SCOPES)
+                .setDataStoreFactory(new FileDataStoreFactory(new java.io.File(TOKENS_DIRECTORY_PATH)))
+                .setAccessType("offline")
+                .build();
+        return new AuthorizationCodeInstalledApp(flow, new LocalServerReceiver()).authorize("user");
+    }
+
+    public static void updateSpreadSheet(List<Object> array) throws Exception {
+        final NetHttpTransport HTTP_TRANSPORT = GoogleNetHttpTransport.newTrustedTransport();
+        final String spreadsheetId = "1fLBMr6hGuCQAGQcljKH_UKaVJBUWm7PH6nEzIjNVla4";
```
Comment (nkt) [inline src/main/java/im/dlg/botsdk/formfillingbot/GoogleSheet.java]: This should be configured

File: src/main/resources/credentials.json [hunk 6]
```diff
@@ -0,0 +1 @@
+{"installed":{"client_id":"630305734862-pu0jco4apns8stp65aa7khlfjl7fb2ap.apps.googleusercontent.com","project_id":"orderlist-1539425836814","auth_uri":"https://accounts.google.com/o/oauth2/auth","token_uri":"https://www.googleapis.com/oauth2/v3/token","auth_provider_x509_cert_url":"https://www.googleapis.com/oauth2/v1/certs","client_secret":"GSYMo0m_3DVCQk4E3SEyCPXa","redirect_uris":["urn:ietf:wg:oauth:2.0:oob","http://localhost"]}}
```
Comment (nkt) [inline src/main/resources/credentials.json]: You shouldn't commit any credentials.

**Comment by m8928** [comment]
still working

**Comment by m8928** [comment]
config.json example
https://pastebin.com/DMpr1GCp

**Comment by Maay** [comment]
@nkt please verify

**Comment by nkt** [comment]
Great, thanks!

**Pull request merged**